plugins/org.jkiss.dbeaver.ext.ui.eula/src/org/jkiss/dbeaver/ext/ui/eula/EULAConfirmationDialog.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.ui.eula;
+
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Shell;
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.runtime.DBWorkbench;
+
+import java.util.prefs.Preferences;
+
+public class EULAConfirmationDialog extends EULABaseDialog {
+    public static final String DBEAVER_EULA = "DBeaver.eula";
+
+    public EULAConfirmationDialog(@NotNull Shell parentShell, @Nullable String eula) {
+        super(parentShell, eula);
+    }
+
+    @Override
+    protected void createButtonsForButtonBar(@NotNull Composite parent) {
+        createButton(parent, IDialogConstants.YES_ID, EULAMessages.core_eula_dialog_accept, false);
+        createButton(parent, IDialogConstants.NO_ID, IDialogConstants.CANCEL_LABEL, false);
+    }
+
+
+    @Override
+    protected boolean canHandleShellCloseEvent() {
+        //We don't want user to close this window
+        return false;
+    }
+
+
+    @Override
+    protected void buttonPressed(int buttonId) {
+        switch (buttonId) {
+            case IDialogConstants.NO_ID:
+                System.exit(101);
+                break;
+            case IDialogConstants.YES_ID:
+                Preferences preferences = Preferences.userNodeForPackage(DBWorkbench.getPlatform().getApplication().getClass());
+                preferences.put(DBEAVER_EULA, EULAUtils.getEulaVersion());
```

Comment:
Consider using `DBWorkbench.getPlatform().getPreferenceStore()` instead of Java Preferences API.